### DISCUSSION TITLE
Get render-json to show values of local variables from included parent

### DISCUSSION THREAD

**Discussion by d-shi** [gruntwork-io/knowledge-base#819] [Help]
Hi,

I'm relatively new to terragrunt so apologies if this has already been answered. I am trying to debug values in local variables, but am currently unable to tell what values they are being set to.

I have a directory with a `terragrunt.hcl` that contains:

```
include {
  path   = find_in_parent_folders("global.terragrunt.hcl")
  expose = true
}
```

The `global.terragrunt.hcl` file is many directory levels up from the directory I'm running terragrunt from. That file contains:

```
locals {
  ...
  account_vars = read_terragrunt_config(find_in_parent_folders("account.hcl"))
  region_vars  = read_terragrunt_config(find_in_parent_folders("region.hcl"))
  ...
}
```

I'm trying to see what some of the local variables in the global file are being set to when I run terragrunt to debug, but when I do `terragrunt render-json` I am getting null for the locals field. This thread seems to indicate that is expected: https://github.com/gruntwork-io/terragrunt/issues/2189

However, when I do `terragrunt render-json --with-metadata` as suggested in that thread, there isn't even a locals field in the output at all.

I also cannot run from the parent directory as suggested since the global file contains lots of `find_in_parent_folders` which is relative to the child directory.

What is the best way for me to debug this?

Thank you!

---

<ins datetime="2024-04-24T21:46:06Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110900">Tracked in ticket #110900</a></p>
</ins>

**Comment by denis256** [comment]
> Currently you can only reference locals defined in the same config file. terragrunt does not automatically include locals defined in the parent config of an include block into the current context

https://terragrunt.gruntwork.io/docs/features/locals/#including-globally-defined-locals

**Comment by d-shi** [comment]
I've read that doc that you linked, but not sure I understand what it's saying. I've inherited this codebase and the original creators are no longer around. For example I have a `terragrunt.hcl` like this:

```
include {
  path   = find_in_parent_folders("global.terragrunt.hcl")
  expose = true
}

terraform {
  source = "../../../../..//modules/vpc/network/v1"
}

inputs = {
  name       = basename(dirname(get_original_terragrunt_dir()))
  aws_region = include.locals.aws_region
}
```

The `global.terragrunt.hcl` is many many levels above where this `terragrunt.hcl` is, and this code works and `aws_region` is successfully used as an input to the `network/v1` module. This is a fairly straightforward example, but I'm asking how I can debug how `include.locals.aws_region` (and other `include.locals`) are rendered.

Hope that makes sense. Thank you!
